File: src/main/java/xyz/nietongxue/serviceql/RootResolver.java
package xyz.nietongxue.serviceql;

import com.coxautodev.graphql.tools.GraphQLQueryResolver;
import org.springframework.stereotype.Component;

import java.util.List;


public class RootResolver implements GraphQLQueryResolver {

    private BookRepository bookRepository;

    public RootResolver(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public List<Book> books() {
        return bookRepository.findAll();
    }
}

File: src/main/kotlin/xyz/nietongxue/serviceql/book/RootResolver.kt
package xyz.nietongxue.serviceql.book


import com.coxautodev.graphql.tools.GraphQLQueryResolver
import org.springframework.stereotype.Component

@Component
class RootResolver(private val bookRepository: BookRepository, private  val authorRepository: AuthorRepository) : GraphQLQueryResolver {

    fun books(): List<Book> {
        return bookRepository.findAll()
    }
    fun authors() = authorRepository.findAll()
}